presto-main/src/main/java/com/facebook/presto/sql/gen/JoinCompiler.java

```diff
@@ -95,11 +110,34 @@
             .build(CacheLoader.from(key ->
                     internalCompileHashStrategy(key.getTypes(), key.getOutputChannels(), key.getJoinChannels(), key.getSortChannel())));
 
+    private final LoadingCache<Type, ScalarFunctionImplementation> distinctFromOperatorCache = CacheBuilder.newBuilder()
+            .recordStats()
+            .maximumSize(1000)
+            .build(CacheLoader.from(this::getOperator));
+
+    private ScalarFunctionImplementation getOperator(Type key)
+    {
+        Signature signature = registry.resolveOperator(OperatorType.IS_DISTINCT_FROM, ImmutableList.of(key, key));
+        return registry.getScalarFunctionImplementation(signature);
+    }
+
     public LookupSourceSupplierFactory compileLookupSourceFactory(List<? extends Type> types, List<Integer> joinChannels, Optional<Integer> sortChannel)
     {
         return compileLookupSourceFactory(types, joinChannels, sortChannel, Optional.empty());
     }
 
+    public JoinCompiler()
```

Comment:
`@Deprecated`?

Reply:
Fair point. But instead of marking it as `@Deprecated`, I recommend inlining it.

@elonazoulay You can do it in a follow up commit. IntelliJ should be able to do this for you automatically.